packages/core/src/components/tag-input/tagInput.tsx

```diff
@@ -184,14 +184,13 @@ export class TagInput extends AbstractPureComponent<ITagInputProps, ITagInputSta
         addOnBlur: false,
         addOnPaste: true,
         inputProps: {},
-        inputValue: "",
```

Comment:
i prefer one fully type-safe `defaultProps` over two not-strictly-typesafe boolean coercions `this.props.inputValue || ""`

Reply:
If I switch this back, I won't be able to explicitly check in `addTags` if `inputValue` is controlled, and will instead have to check if `inputValue === ""`. I think that works in this case, but is less explicit. Would you still prefer me to set the `inputValue` default here?

Reply:
oh hmm that's a thing. in other components, i've checked `=== X.defaultProps.Y` to great success.